src/merge_tools.rs

```diff
@@ -432,52 +443,60 @@ fn get_tool_config(settings: &UserSettings, name: &str) -> Result<MergeTool, Con
         if result.program.is_empty() {
             result.program.clone_from(&name.to_string());
         };
-        Ok(result)
+        Ok(Some(result))
     } else {
-        Ok(MergeTool::with_program(name))
+        Ok(None)
     }
 }
 
 fn get_diff_editor_from_settings(
     ui: &mut Ui,
     settings: &UserSettings,
 ) -> Result<MergeTool, ExternalToolError> {
-    let editor_name = editor_name_from_settings(ui, settings, "ui.diff-editor")?;
-    Ok(get_tool_config(settings, &editor_name)?)
+    let args = editor_args_from_settings(ui, settings, "ui.diff-editor")?;
+    let maybe_editor = match &args {
+        FullCommandArgs::String(name) => get_tool_config(settings, name)?,
```

Comment:
I think "FullCommandArgs" is a bit confusing. We should rename it to "ExternalToolCommandArgs" or something like that (not necessarily in this PR).

Reply:
I agree "FullCommandArgs" isn't the best name, but it's not specific to external tools.
It's a command `name + args` constructed from string or array.

Reply:
Maybe `CommandWithArgs`?

Reply:
or CommandNameWithArgs? Not sure if it's better than "full" "args" because "args" may include command name depending on context.

Reply:
`CommandNameAndArgs` is a bit better, I think. I thought that the name is implicit in `CommandWithArgs`, but if you don't think so, that proves it's a potential point of confusion.